// cos-wx-sdk-v5/demo/37-yunchi-taro/src/app.service.js
import Taro from '@tarojs/taro'
import { get as getGlobalData, set as setGlobalData } from './app.global'

var service = {
    upsert: (params) => new Promise((resolve, reject) => {
        Taro.request({
            url: getGlobalData('httpUrl') + 'user/upsert',
            method: 'POST',
            data: {
                ...params,
                openid: getGlobalData('openid'),
                session_key: getGlobalData('session_key'),
            },
            header: {
                'Authorization': getGlobalData('token')
            },
            success: (res) => resolve(res),
            fail: (res) => reject(res)
        })
    }),
    login: () => new Promise((resolve, reject) => {
        Taro.login({
            success: res0 => {
                Taro.request({
                    url: getGlobalData('httpUrl') + 'wx/login',
                    method: 'POST',
                    data: {
                        code: res0.code,
                        appid: getGlobalData('appid'),
                        secret: getGlobalData('appkey'),
                    },
                    header: {
                        'Authorization': getGlobalData('token')
                    },
                    success: res1 => {
                        setGlobalData('openid', res1.data.openid)
                        setGlobalData('session_key', res1.data.session_key)
                        resolve(res1)
                        // Taro.request({
                        //     url: getGlobalData('httpUrl') + 'session/upsert',
                        //     method: 'POST',
                        //     data: {
                        //         openid: res1.data.openid,
                        //         session_key: res1.data.session_key,
                        //     },
                        //     header: {
                        //         'Authorization': getGlobalData('token')
                        //     },
                        //     success: () => resolve(res1),
                        //     fail: (res) => reject(res)
                        // })
                    },
                    fail: (res) => reject(res)
                })
            }
        })
    }),
    userLogin: (params) => new Promise((resolve, reject) => {
        Taro.request({
            url: getGlobalData('httpUrl') + 'user/login',
            method: 'POST',
            data: {
                ...params,
                openid: getGlobalData('openid'),
                session_key: getGlobalData('session_key'),
            },
            header: {
                'Authorization': getGlobalData('token')
            },
            success: (res) => resolve(res),
            fail: (res) => reject(res)
        })
    }),



}

module.exports = service;
